frontend/src/queries/nodes/DataVisualization/Components/Variables/Variables.tsx

```diff
@@ -1,81 +1,141 @@
-                            {
-                                title: 'New variable',
-                                items: [
-                                    {
-                                        label: 'String',
-                                        onClick: () => openModal('String'),
-                                    },
-                                    {
-                                        label: 'Number',
-                                        onClick: () => openModal('Number'),
-                                    },
-                                    {
-                                        label: 'Boolean',
-                                        onClick: () => openModal('Boolean'),
-                                    },
-                                    {
-                                        label: 'List',
-                                        onClick: () => openModal('List'),
-                                    },
-                                ],
-                            },
-                            {
-                                label: 'Existing variable',
-                                items: variablesLoading
-                                    ? [
-                                          {
-                                              label: 'Loading...',
-                                              onClick: () => {},
-                                          },
-                                      ]
-                                    : variables.map((n) => ({
-                                          label: n.name,
-                                          onClick: () => addVariable(n.id),
-                                      })),
-                            },
-                        ]}
-                    >
-                        <LemonButton type="secondary" sideIcon={<IconPlus />}>
-                            Add variable
-                        </LemonButton>
-                    </LemonMenu>
-                )}
+            <div className="flex gap-4 flex-wrap px-px">
                 {variablesForInsight.map((n) => (
-                    <div key={n.id}>{n.name}</div>
+                    <VariableComponent key={n.id} variable={n} />
                 ))}
             </div>
             <NewVariableModal />
         </>
     )
 }
+
+const VariableInput = ({ variable, closePopover }: { variable: Variable; closePopover: () => void }): JSX.Element => {
+    const { showEditingUI } = useValues(dataVisualizationLogic)
+    const { updateVariableValue } = useActions(variablesLogic)
+
+    const [localInputValue, setLocalInputValue] = useState(variable.value ?? variable.default_value ?? '')
+
+    const inputRef = useRef<HTMLInputElement>(null)
+    const codeRef = useRef<HTMLElement>(null)
+
+    useEffect(() => {
+        inputRef.current?.focus()
+    }, [inputRef.current])
+
+    const variableAsHogQL = `{variables.${variable.code_name}}`
```

Comment:
If you change this:

```suggestion
    import { hogql } from 'queries/utils'
    const variableAsHogQL = `{variables.${hogql.identifier(variable.code_name)}}`
```

Then you don't need to sanitize the code name. And even if you sanitize, it's good practice to escape it anyway --> who knows what data might end up in the database.

Also look in utils.test.ts to see other usages of this `hogql` templating system, such as:

```ts
    it('properly returns query with string array substitution', () => {
        expect(hogql`SELECT * FROM events WHERE properties.foo IN ${['bar', 'baz']}`).toEqual(
            "SELECT * FROM events WHERE properties.foo IN ['bar', 'baz']"
        )
    })
```

Reply:
This is actually dictated on the backend and stored - we create `code_name` from the variable name (with some char stripping). Looking at the backend version of `escape_hogql_identifier` it doesn't look like it'll do what I want, so I think this is better off as-is